### DISCUSSION TITLE
MQTT.Simple wait_msg throws OSError: -28928

### DISCUSSION THREAD

**Discussion by michaeka79** [micropython/micropython#9143] [Libraries & Drivers] [answered]
Hi!

I'm trying to get MQTT messaging to work on my Pico W. However, I just ran into an issue where MQTT.Simple is throwing a OSError -28928 when I go to enter an IF statement in the callback. This was working for me for a minute then starting throwing the error for seemingly no reason.   Any thoughts on what might be happening? I removed some stuff from the  IF thinking it could be memory related but that didn't work either.

**Comment by davefes** [comment]
[https://forums.mbed.com/t/dtls-fallback-mechanism-or-faulty-config/8073](url)
... maybe a SSL issue with MBEDTLS.  Does the Pico W use MBEDTLS?

**Comment by michaeka79** [comment, marked as answer]
WOW! That was a WEIRD error!

This error was due to the incorrect variable type being compared in an IF statement. When the code got back around again to check the wait message it was in some error state which blew up the MQTT client.
